src/test/java/org/jitsi/meet/test/mobile/base/MobileParticipantOptions.java

```diff
@@ -15,13 +15,215 @@
+        = _CAPS_PROP_PREFIX + "platformName";
+
+    /**
+     * Default Appium server address (local).
+     */
+    private static final String DEFAULT_APPIUM_SERVER_ADDRESS = "127.0.0.1";
+
+    /**
+     * Default Appium server port.
+     */
+    private static final String DEFAULT_APPIUM_SERVER_PORT = "4723";
+
+    /**
+     * The name of the property which specifies the Appium server address.
+     */
+    private static final String PROP_APPIUM_SERVER_ADDRESS
+        = "appium.server.address";
+
+    /**
+     * The name of the property which specifies the Appium server's port.
+     */
+    private static final String PROP_APPIUM_SERVER_PORT = "appium.server.port";
+
+    /**
+     * The property name for bundle ID. Required to identify reinstalled app.
+     * @see #PROP_REINSTALL_APP
+     */
+    static final String PROP_BUNDLE_ID = "bundleId";
+
+    /**
+     * The property key which when set to "true" will reinstall the app, before
+     * the test will start.
+     */
+    private static final String PROP_REINSTALL_APP = "reinstallApp";
+
+    /**
+     * Creates {@link DesiredCapabilities} out of the capabilities properties
+     * set held by this instance (all that start with
+     * {@link #_CAPS_PROP_PREFIX}).
+     *
+     * @return new {@link DesiredCapabilities} instance configured based on
+     * the capabilities options held by this instance.
+     */
+    public DesiredCapabilities createCapabilities()
+    {
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+
+        readAndSetCapability(capabilities, CAPS_APP);
+        readAndSetCapability(capabilities, CAPS_PLATFORM_NAME);
+        readAndSetCapability(capabilities, CAPS_DEVICE_NAME);
+
+        // This is required to not timeout drivers waiting for other devices
+        // to start
+        capabilities.setCapability("newCommandTimeout", 120);
+
+        return capabilities;
+    }
+
+    /**
+     * The getter for {@link #PROP_APPIUM_SERVER_ADDRESS}.
+     *
+     * @return the server address part of the URL.
+     */
+    private String getAppiumServerAddress()
+    {
+        return getProperty(PROP_APPIUM_SERVER_ADDRESS);
+    }
+
+    /**
+     * The getter for {@link #PROP_APPIUM_SERVER_PORT}.
+     *
+     * @return the port number as a String.
+     */
+    private String getAppiumServerPort()
+    {
+        return getProperty(PROP_APPIUM_SERVER_PORT);
+    }
+
+    /**
+     * Constructs Appium server URL.
+     *
+     * @return {@link URL} from the options held by this instance.
+     */
+    public URL getAppiumServerUrl()
+    {
+        String appiumAddress
+            = Objects.requireNonNull(
+                    getAppiumServerAddress(), "appiumServerAddress");
+        String appiumPort
+            = Objects.requireNonNull(
+                    getAppiumServerPort(), "appiumServerPort");
+
+        try
+        {
+            return new URL(
+                "http://" + appiumAddress + ":" + appiumPort + "/wd/hub");
```

Comment:
is http instead of https on purpose?

Reply:
Yep, connections to the selenium grid are http.

Reply:
I'm not sure if https will work, but will try that and change to http or even go with the full address if possible, but I think it was split into address and port because of Amazon device farm which provides those 2 in system properties